pkg/pkgmgr/rpm.go

```diff
@@ -81,8 +82,22 @@ func (st rpmDBType) String() string {
 
 // Depending on go-rpm-version lib for RPM version comparison rules.
 func isValidRPMVersion(v string) bool { // nolint:revive
-	// TODO: Verify if there are format correctness check that need to be added given lack of support in rpmVer lib
-	return true
+	err := isValidPackage(v)
+	return err == nil
+}
+
+func isValidPackage(ver string) error {
+	if !unicode.IsDigit(rune(ver[0])) {
+		return errors.New("upstream_version must start with digit")
+	}
+
+	allowedSymbols := ".-+~:_"
+	for _, s := range ver {
+		if !unicode.IsDigit(s) && !unicode.IsLetter(s) && !strings.ContainsRune(allowedSymbols, s) {
+			return errors.New("upstream_version includes invalid character")
+		}
```

Comment:
Using the strings.ContainsAny() function might be more concise here. Also maybe we can log the upstream_version in the error message. Otherwise the checks look good to me based on the docs!